src/lib.rs

```diff
@@ -709,22 +708,22 @@ impl<'s> Parser<'s> {
                         inline::Container::Mark => Container::Mark,
                         inline::Container::InlineLink => Container::Link(
                             match self.inlines.src(inline.span) {
-                                CowStr::Owned(s) => s.replace('\n', "").into(),
                                 s @ CowStr::Borrowed(_) => s,
+                                s => s.replace('\n', "").into(),
```

Comment:
would be nice to avoid allocation when we have an inline here. I guess a custom filter function would be needed.